File: OneDrive/Documentos/Store_Herencia/Store_Herencia/src/store/herencia/Product.java
package store.herencia;

public class Product{
    private static int CANTIDAD_PRODUCTOS = 0;
    private int serialNumber;
    private String name;
    private double price;
    private String importDate;
    private int stock;

    public Product(String name, double price, String importDate, int stock) {
        this.serialNumber = CANTIDAD_PRODUCTOS++;
        this.name = name;
        this.price = price;
        this.importDate = importDate;
        this.stock = stock;
        CANTIDAD_PRODUCTOS++;
    }

    protected String getData() {
        return String.format("Nombre: %s, precio %f, fecha de importación: %s, stock: %d",
                name, price, importDate, stock);
    }
    public int getSerialNumber() {
        return serialNumber;
    }

    public void setSerialNumber(int serialNumber) {
        this.serialNumber = serialNumber;
    } 
    
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getImportDate() {
        return importDate;
    }

    public void setImportDate(String importDate) {
        this.importDate = importDate;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }
}

File: PycharmProjects/Store_Herencia/py/Product.py
class Product:
    CANTIDAD_PRODUCTOS = 0

    def __init__(self, name, price, import_date, stock):
        Product.CANTIDAD_PRODUCTOS += 1
        self.serial_number = Product.CANTIDAD_PRODUCTOS
        self.name = name
        self.price = price
        self.import_date = import_date
        self.stock = stock

    def get_data(self):
        return f"Nombre: {self.name}, precio: {self.price}, fecha de importación: {self.import_date}, stock: {self.stock}"

    def get_serial_number(self):
        return self.serial_number

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_price(self):
        return self.price

    def set_price(self, price):
        self.price = price

    def get_import_date(self):
        return self.import_date

    def set_import_date(self, import_date):
        self.import_date = import_date

    def get_stock(self):
        return self.stock

    def set_stock(self, stock):
        self.stock = stock
